pkg/block/fetcher.go

```diff
@@ -428,15 +428,30 @@ func NewDeduplicateFilter() *DeduplicateFilter {
 // Filter filters out duplicate blocks that can be formed
 // from two or more overlapping blocks that fully submatches the source blocks of the older blocks.
 func (f *DeduplicateFilter) Filter(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, _ bool) {
-	root := NewNode(&metadata.Meta{
-		BlockMeta: tsdb.BlockMeta{
-			ULID: ulid.MustNew(uint64(0), nil),
-		},
-	})
+	var wg sync.WaitGroup
+	for _, res := range []int64{
+		int64(0), int64(5 * 60 * 1000), int64(60 * 60 * 1000),
+	} {
+		wg.Add(1)
+		go func(res int64) {
+			defer wg.Done()
+			f.filter(NewNode(&metadata.Meta{
+				BlockMeta: tsdb.BlockMeta{
+					ULID: ulid.MustNew(uint64(0), nil),
+				},
+			}), metas, res, synced)
+		}(res)
+	}
 
+	wg.Wait()
+}
+
+func (f *DeduplicateFilter) filter(root *Node, metas map[ulid.ULID]*metadata.Meta, resolution int64, synced GaugeLabeled) {
```

Comment:
Small suggestion for readability:
```suggestion
func (f *DeduplicateFilter) filterForResolution(root *Node, metas map[ulid.ULID]*metadata.Meta, res int64, synced GaugeLabeled) {
```